qlib/workflow/task/gen.py

```diff
@@ -194,6 +196,16 @@ def generate(self, task: dict):
 
             # update segments of this task
             t["dataset"]["kwargs"]["segments"] = copy.deepcopy(segments)
+            # if end_time < the end of test_segments, then change end_time to allow load more data
+            if (
+                self.modify_end_time
+                and self.ta.cal_interval(
+                    t["dataset"]["kwargs"]["handler"]["kwargs"]["end_time"],
```

Comment:
Handler is not a necessary part of the dataset.


end_time is not ensured to appear in kwargs